Libraries/LibWeb/HTML/Canvas/CanvasDrawImage.cpp

```diff
@@ -62,6 +66,38 @@ static void default_source_size(CanvasImageSource const& image, float& source_wi
         });
 }
 
+Gfx::ImageOrientation get_image_orientation_from_canvas_source(CanvasImageSource const& source)
+{
+    auto const image_orientation = source.visit(
+        [](GC::Root<HTMLImageElement> const& source) -> CSS::ImageOrientation {
+            auto const& computed_properties = source->computed_properties();
+            return computed_properties
+                ? computed_properties->image_orientation()
+                : CSS::ImageOrientation::FromImage;
+        },
+        [](GC::Root<SVG::SVGImageElement> const& source) -> CSS::ImageOrientation {
+            auto const& computed_properties = source->computed_properties();
+            return computed_properties
+                ? computed_properties->image_orientation()
+                : CSS::ImageOrientation::FromImage;
+        },
+        [](GC::Root<HTMLCanvasElement> const& source) -> CSS::ImageOrientation {
+            auto const& computed_properties = source->computed_properties();
+            return computed_properties
+                ? computed_properties->image_orientation()
+                : CSS::ImageOrientation::FromImage;
+        },
+        [](GC::Root<HTMLVideoElement> const& source) -> CSS::ImageOrientation {
+            auto const& computed_properties = source->computed_properties();
+            return computed_properties
+                ? computed_properties->image_orientation()
+                : CSS::ImageOrientation::FromImage;
+        },
+        [](GC::Root<ImageBitmap> const&) -> CSS::ImageOrientation { return CSS::ImageOrientation::FromImage; });
```

Comment:
As most of these are the same code, a clever trick you can use is to put `auto const&` as the parameter (as long as it's after all the ones with explicit types). eg:

```suggestion
    auto const image_orientation = source.visit(
        [](GC::Root<ImageBitmap> const&) -> CSS::ImageOrientation { return CSS::ImageOrientation::FromImage; },
        [](auto const& source) -> CSS::ImageOrientation {
            auto const& computed_properties = source->computed_properties();
            return computed_properties
                ? computed_properties->image_orientation()
                : CSS::ImageOrientation::FromImage;
        });
```